frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/ReverificationResultHandlerTest.java

```diff
@@ -0,0 +1,170 @@
+    void setUp() throws URISyntaxException {
+        when(context.getAwsRequestId()).thenReturn("aws-session-id");
+        when(sessionService.getSessionFromRequestHeaders(anyMap()))
+                .thenReturn(Optional.of(session));
+        when(configurationService.getIPVBackendURI())
+                .thenReturn(new URI("https://api.identity.account.gov.uk/token"));
+
+        handler =
+                new ReverificationResultHandler(
+                        configurationService,
+                        sessionService,
+                        clientSessionService,
+                        clientService,
+                        authenticationService,
+                        reverificationResultService,
+                        auditService);
+    }
+
+    @Test
+    void shouldReturn200AndIPVResponseOnValidRequest()
+            throws ParseException, UnsuccessfulReverificationResponseException {
+        HTTPResponse userInfo = new HTTPResponse(200);
+        userInfo.setContent(
+                "{ \"sub\": \"urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6\",\"success\": true}");
+        when(reverificationResultService.getToken(any())).thenReturn(getSuccessfulTokenResponse());
+        when(reverificationResultService.sendIpvReverificationRequest(any())).thenReturn(userInfo);
+
+        var result = handler.handleRequest(apiRequestEventWithEmail("1234", EMAIL), context);
+
+        assertThat(result, hasStatus(200));
+        assertThat(result, hasBody(userInfo.getContent()));
+    }
+
+    @Test
+    void shouldHandleIPVTokenError()
+            throws ParseException, UnsuccessfulReverificationResponseException {
+        when(reverificationResultService.getToken(any()))
+                .thenReturn(getUnsuccessfulTokenResponse());
+        when(reverificationResultService.sendIpvReverificationRequest(any()))
+                .thenThrow(
+                        new UnsuccessfulReverificationResponseException(
+                                "Error getting reverification result"));
+
+        var result = handler.handleRequest(apiRequestEventWithEmail("1234", EMAIL), context);
+
+        assertThat(result, hasStatus(400));
+        assertThat(result, hasJsonBody(ERROR_1058));
+    }
+
+    @Test
+    void shouldHandleIPVReverificationError()
+            throws ParseException, UnsuccessfulReverificationResponseException {
+        when(reverificationResultService.getToken(any())).thenReturn(getSuccessfulTokenResponse());
+        when(reverificationResultService.sendIpvReverificationRequest(any()))
+                .thenThrow(
+                        new UnsuccessfulReverificationResponseException(
+                                "Error getting reverification result"));
+
+        var result = handler.handleRequest(apiRequestEventWithEmail("1234", EMAIL), context);
+
+        assertThat(result, hasStatus(400));
+        assertThat(result, hasJsonBody(ERROR_1059));
+    }
+
+    private Map<String, String> getHeaders() {
```

Comment:
You can remove this method and just use VALID_HEADERS directly from common test variables